File: examples/vulnerable_crypto.py
"""Example file with quantum-vulnerable cryptography for testing."""

from cryptography.hazmat.primitives.asymmetric import rsa, ec, dsa
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import padding

# RSA key generation - QUANTUM VULNERABLE
def generate_rsa_key():
    """Generate RSA private key."""
    private_key = rsa.generate_private_key(
        public_exponent=65537,
        key_size=2048,  # Will be flagged as vulnerable
    )
    return private_key

# Weak RSA key - CRITICAL
def generate_weak_rsa():
    """Generate weak RSA key."""
    private_key = rsa.generate_private_key(
        public_exponent=65537,
        key_size=1024,  # CRITICAL - too small
    )
    return private_key

# ECC key generation - QUANTUM VULNERABLE  
def generate_ecc_key():
    """Generate ECC private key."""
    private_key = ec.generate_private_key(ec.SECP256R1())
    return private_key

# DSA key generation - QUANTUM VULNERABLE
def generate_dsa_key():
    """Generate DSA private key.""" 
    private_key = dsa.generate_private_key(
        key_size=2048
    )
    return private_key

# RSA encryption example
def rsa_encrypt_data(public_key, data):
    """Encrypt data with RSA public key."""
    encrypted = public_key.encrypt(
        data,
        padding.OAEP(
            mgf=padding.MGF1(algorithm=hashes.SHA256()),
            algorithm=hashes.SHA256(),
            label=None
        )
    )
    return encrypted

# ECDSA signing example
def sign_with_ecdsa(private_key, data):
    """Sign data with ECDSA private key."""
    signature = private_key.sign(
        data,
        ec.ECDSA(hashes.SHA256())
    )
    return signature

if __name__ == "__main__":
    # Demo usage - all quantum vulnerable!
    rsa_key = generate_rsa_key()
    weak_rsa = generate_weak_rsa()
    ecc_key = generate_ecc_key()
    dsa_key = generate_dsa_key()
    
    print("Generated quantum-vulnerable keys!")
    print("These will be broken by quantum computers!")

File: examples/vulnerable_crypto.go
// Example Go file with quantum-vulnerable cryptography for testing
package main

import (
	"crypto/rand"
	"crypto/rsa"
	"crypto/ecdsa"
	"crypto/elliptic"
	"fmt"
	"log"
)

// generateRSAKey generates RSA private key - QUANTUM VULNERABLE
func generateRSAKey() (*rsa.PrivateKey, error) {
	return rsa.GenerateKey(rand.Reader, 2048) // Will be flagged as vulnerable
}

// generateWeakRSAKey generates weak RSA key - CRITICAL
func generateWeakRSAKey() (*rsa.PrivateKey, error) {
	return rsa.GenerateKey(rand.Reader, 1024) // CRITICAL - too small
}

// generateECDSAKey generates ECDSA private key - QUANTUM VULNERABLE
func generateECDSAKey() (*ecdsa.PrivateKey, error) {
	return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
}

// generateECDSAKeyP384 generates ECDSA P-384 key - QUANTUM VULNERABLE
func generateECDSAKeyP384() (*ecdsa.PrivateKey, error) {
	return ecdsa.GenerateKey(elliptic.P384(), rand.Reader)
}

func main() {
	// Demo usage - all quantum vulnerable!
	rsaKey, err := generateRSAKey()
	if err != nil {
		log.Fatal(err)
	}
	
	weakRSAKey, err := generateWeakRSAKey()
	if err != nil {
		log.Fatal(err)
	}
	
	ecdsaKey, err := generateECDSAKey()
	if err != nil {
		log.Fatal(err)
	}
	
	ecdsaP384Key, err := generateECDSAKeyP384()
	if err != nil {
		log.Fatal(err)
	}
	
	fmt.Printf("Generated RSA key: %v\n", rsaKey != nil)
	fmt.Printf("Generated weak RSA key: %v\n", weakRSAKey != nil)
	fmt.Printf("Generated ECDSA key: %v\n", ecdsaKey != nil)
	fmt.Printf("Generated ECDSA P-384 key: %v\n", ecdsaP384Key != nil)
	
	fmt.Println("All generated keys are quantum vulnerable!")
	fmt.Println("These will be broken by quantum computers!")
}